numalogic/udfs/tools.py

```diff
@@ -57,6 +60,24 @@ def get_df(
     return df[features].astype(np.float32), df["timestamp"].astype(int).tolist()
 
 
+def _update_info_metric(
+    data: np.ndarray, metric_name: Sequence[str], labels: Sequence[str]
+) -> None:
+    """
+    Utility function is used to update the gauge metric.
+    Args:
+        data: data
+        metric_name: metric name in the payload
+        labels: labels.
+    """
+    for _data, _metric_name in zip(data.T, metric_name):
+        _set_gauge(
+            gauge=RECORDED_DATA_GAUGE,
+            labels=(*labels, _metric_name),
+            data=np.mean(_data).squeeze(),
+        )
+
```

Comment:
Can you explain what are we trying to do here?

Reply:
This basically takes the mean of the data metric wise.